### DISCUSSION TITLE
Usercentrics Cookie Consent Banner Not Displayed for YouTube Videos in Glightbox

### DISCUSSION THREAD

**Discussion by JS-Media-Creation** [biati-digital/glightbox#495] [General]
Hello,

I am using Glightbox to embed YouTube videos on my website. Typically, when a user has previously declined the necessary cookies for YouTube, a cookie consent banner from Usercentrics appears, allowing the user to consent and watch the video.

However, I noticed that when embedding YouTube videos within a Glightbox, the Usercentrics cookie consent banner is not displayed. Instead, the video fails to load and a loading spinner appears indefinitely. This issue prevents users from granting the required cookie permissions to watch the video.

I am unsure whether this issue is related to Glightbox or if it stems from the Usercentrics cookie consent tool. It is worth noting that the banner is embedded via Shadow Root, which makes direct manipulation challenging.

Could you please investigate this issue and consider adding support for displaying the Usercentrics cookie consent banner when a YouTube video is embedded in a Glightbox? This would ensure that users are prompted to give the necessary permissions and can view the video without any issues.

Thank you for your attention to this matter.

Steps to Reproduce:

	1.	Embed a YouTube video using Glightbox.
	2.	Ensure that Usercentrics cookie consent is required for YouTube videos.
	3.	Decline the necessary cookies.
	4.	Attempt to open the video in the Glightbox.

Expected Behavior:
The Usercentrics cookie consent banner should be displayed, allowing users to consent to the necessary cookies and watch the video.

Actual Behavior:
The video fails to load, and a loading spinner is displayed indefinitely.

Environment:

	•	Glightbox version: 3.2.0
	•	Browser: All

Example:

	•	https://skillxprobe.de

Thank you for your help!

Best regards,
Jonathan

**Comment by biati-digital** [comment]
I've never user Usercentrics and not sure someone will invest their time investigating and doing tests but I'll keep the discussion open in case someone knows a workaround or already have a solution.

**Comment by JS-Media-Creation** [comment]
I have observed that when the Lightbox, which continuously attempts to load the video without success, is closed, the Usercentrics cookie consent banner briefly appears for a moment and then disappears together with the video thumbnail. Maybe this fact can help to find a workaround or solution.

<img width="1445" alt="Bildschirmfoto 2024-06-19 um 12 55 33" src="https://github.com/biati-digital/glightbox/assets/50705044/b27a45ff-cc9f-46b2-b553-c46924a346f0">
<img width="1443" alt="Bildschirmfoto 2024-06-19 um 12 58 43" src="https://github.com/biati-digital/glightbox/assets/50705044/dcb977dc-85ac-40e0-8572-e9df652d8a70">
